cds_ils/importer/importer.py

```diff
@@ -288,3 +288,62 @@ def preview_import(self):
             self.action = None
 
         return self.import_summary()
+
+    def preview_delete_document(self, document):
+        """Delete Document record."""
+        loan_search_res = document.search_loan_references()
+        item_search_res = document.search_item_references()
+        req_search_res = document.search_doc_req_references()
+        orders_refs_search = document.search_order_references()
+        brw_req_refs_search = document.search_brw_req_references()
+
+        if loan_search_res.count():
+            raise CDSRecordHasReferencesError(
+                document=document,
+                ref_type="Loan",
+                refs=loan_search_res,
+            )
+
+        if item_search_res.count():
+            raise CDSRecordHasReferencesError(
+                document=document,
+                ref_type="Item",
+                refs=item_search_res,
+            )
+
+        if req_search_res.count():
+            raise CDSRecordHasReferencesError(
+                document=document,
+                ref_type="DocumentRequest",
+                refs=req_search_res,
+            )
+
+        if orders_refs_search.count():
+            raise CDSRecordHasReferencesError(
+                document=document,
+                ref_type="AcquisitionOrder",
+                refs=orders_refs_search,
+            )
+
+        if brw_req_refs_search.count():
+            raise CDSRecordHasReferencesError(
+                document=document,
+                ref_type="BorrowingRequest",
+                refs=brw_req_refs_search,
+            )
+
+        related_refs = set()
+        for _, related_objects in document.relations.items():
+            for obj in related_objects:
+                if not obj["record_metadata"]["mode_of_issuance"] == "SERIAL":
+                    related_refs.add("{pid_value}:{pid_type}".format(**obj))
+        if related_refs:
+            raise RecordHasReferencesError(
+                record_type=document.__class__.__name__,
+                record_id=document["pid"],
+                ref_type="related",
+                ref_ids=sorted(ref for ref in related_refs),
+            )
+
+        if document.model is None:
```

Comment:
not sure why you are checking the model here ?

Reply:
It's just another check that happens when you try to delete an item so, if it can happen in the real delete, it should also be possible in the preview.

Reply:
not sure if I understand what you mean, lets discuss in person

Reply:
I am not sure about this either :/

Reply:
My rational here is the following:
When we try to delete a document we call to the delete method of invenio_app_ils/documents/api/Document which at the end calls the delete function of it's parent invenio_app_ils/records_relations/api/IlsRecordWithRelations which also calls in the end the function method of it's parent invenio_records/api/Record. In this last delete function there is a condition stating that if the model of the record is None a MissingModelError exception should be raised.
I don't think it could happen in our case but, if we want the preview to make the same checks than the real delete and raise the same errors, I think this should be added.
Regardless, this is just what I think and I don't feel confident stating this should be kept.

Reply:
lets discuss this in a call as soon as possible